README.md

```diff
@@ -1,107 +1,149 @@
+counter keeps track of the amount of jmp terms, and the second counter
+is incremented every time we enter any term.  The `main` function, just
+runs the counter. We register it with `Project.register_pass'`, and our
+function will be called by the system, as soon as project is loaded and
+ready for the analysis.
 
-```bash
-$ bapbuild mycoolprog.native
+To compile the plugin simply run the following command:
+
+```
+bapbuild jmp.plugin
 ```
 
-If `bapbuild` complains that something is missing, make sure that you
-didn't skip the [Installation](#Installation) phase. You can add your
-own dependencies with a `-pkg` or `-pkgs` command line options:
+It is easier to run the pass, if it is installed, so let's do it:
 
-```bash
-$ bapbuild -pkg lwt mycoolprog.native
+```
+bapbundle install jmp.plugin
+```
+
+Now we can test it:
+```
+$ bap /bin/true --pass=jmps
+ratio = 974/7514 = 0.129625
+$ bap /bin/ls --pass=jmps
+ratio = 8917/64557 = 0.138126
 ```
 
-If you use your own build environment, please make sure that you have
-added `bap` as a dependency. We install our libraries using
-`ocamlfind` and you just need to add `bap` to your project. For
-example, if you use `oasis`, then you should add `bap` to the
-`BuildDepends` field. If you are using `ocamlbuild` with the
-`ocamlfind` plugin, then you should add `package(bap)` or `pkg_bap` to
-your `_tags` file.
+## Python
 
+OK, if the previous example doesn't make any sense for you, then you
+can try our
+[Python bindings](https://github.com/BinaryAnalysisPlatform/bap-python).
+Install them with `pip install bap` (you still need to install `bap`
+before hand). Here is the same example, but in Python:
 
-## Using from top-level
+```python
+import bap
+from bap.adt import Visitor
 
-It maybe a good idea to learn how to use our library by playing in an
-OCaml top-level. If you have installed `utop`, then you can just use
-our `baptop` script to run `utop` with `bap` extensions:
+class Counter(Visitor) :
+    def __init__(self):
+        self.jmps = 0
+        self.total = 0
+
+    def enter_Jmp(self,jmp):
+        self.jmps += 1
+
+    def enter_Term(self,t):
+        self.total += 1
+
+proj = bap.run('/bin/true')
+count = Counter()
+count.run(proj.program)
+print("ratio = {0}/{1} = {2}".format(count.jmps, count.total,
+                                     count.jmps/float(count.total)))
+```
+
+
+## baptop
+
+BAP also ships an interactive toplevel, aka REPL. This is a shell-like
+program, that will interactively evaluate OCaml instructions, and
+print results. Just run:
 
 ```bash
 $ baptop
 ```
 
-Now, you can play with BAP. The following example, will create a
-project from
-[coreutils_O2_true](https://github.com/BinaryAnalysisPlatform/arm-binaries/raw/master/coreutils/coreutils_O2_true)
-file, build callgraph of a program, control flow graph and dominance
-tree of a `main` function.
+Now, you can play with BAP. The following example will open a file,
```

Comment:
Maybe bullet these? Dunno

will:
* open a file
* build a program callgraph
* build a control flow graph
* build a dominance tree of a function